share/doc/DEVELOPERS_GUIDE.rst

```diff
@@ -73,7 +73,79 @@ Making a Release
+
+Major releases
+~~~~~~~~~~~~~~
+
+In preparation for a major (not minor!) release the following additional steps need to be done:
+-  Create a branch, e.g. :code:`stable_bump` in :code:`votca/votca` and all submodules from the current master
+   ::
+
+       git checkout master
+       git submodules foreach git checkout master
+       git checkout -b stable_bump
+       git submodules foreach git checkout -b stable_bump
+
+-  Bump the version in CMake to :code:`<major>-rc.1` replacing :code:`<major>-dev` in :code:`votca/votca` and all submodules
+-  Update the :code:`CHANGELOG.rst` files accordingly, by changing the top most section from :code:`<major>-dev` to `:code:`<major>-rc.1`
+-  Commit changes in all submodules and update the submodules in :code:`votca/votca`
+   ::
+
+       git submodules foreach git commit -m "Version bumped to <major>-rc.1"
+       git add -u
+       git commit -m " Version bumped to <major>-rc.1"
+
+-  Push everything, but do NOT make the pull requests yet
+   ::
+
+       git submodules foreach git push origin stable_bump
+       git push origin stable_bump
+
+-  Create a branch, e.g. :code:`master_bump`, in :code:`votca/votca` and all submodules from the current master
+   ::
+
+       git checkout master
+       git submodules foreach git checkout master
+       git checkout -b master_bump
+       git submodules foreach git checkout -b master_bump
+
+-  Bump the version in CMake to :code:`<major>-dev` replacing :code:`<major+1>-dev` in :code:`votca/votca` and all submodules
+-  Create a new secion in the :code:`CHANGELOG.rst` files for :code:`<major+1>-dev`
+-  Commit changes in all submodules and update the submodules in :code:`votca/votca`
+   ::
+
+       git submodules foreach git commit -m "Version bumped to <major+1>-dev"
+       git add -u
+       git commit -m " Version bumped to <major+1>-dev"
+
+-  Push everything, but do NOT make the pull requests yet
+   ::
+
+       git submodules foreach git push origin master_bump
+       git push origin master_bump
+
+-  Now, create a PR in :code:`votca/votca` from :code:`master_bump` into :code:`master`
+-  Once merged, create PRs in all submodules from :code:`master_bump` into :code:`master`
+-  Once all of these are merged and the automatically "Update master submodules" PR is merged, start with :code:`stable_bump` PRs
+-  Create a PR in :code:`votca/votca` from the :code:`stable_bump` branch into the :code:`stable` branch
+-  Once merged, create PRs in all submodules from each of the :code:`stable_bump` branches into each of their :code:`stable` branches
+-  Once all of them are merged and merge the automatically "Update stable submodules" PR
+-  Now everything is ready for the automatic release creation by Github Actions
```

Comment:
I think adding step numbers here would be beneficial or throughout this section because then you can directly refer to the previous steps without it being ambiguous. 

E.g. 

Once all of the PRs from step () are merged do this.

Reply:
The problem is numbers is they always get out of sync, because people never update them.

Reply:
Ok, your call.